chain/messagepool/provider.go

```diff
@@ -53,18 +65,31 @@ func NewProvider(sm *stmgr.StateManager, ps *pubsub.PubSub) Provider {
 }
 
 func NewProviderLite(sm *stmgr.StateManager, ps *pubsub.PubSub, noncer MpoolNonceAPI) Provider {
-	return &mpoolProvider{sm: sm, ps: ps, lite: noncer}
+	return &mpoolProvider{
+		sm:             sm,
+		ps:             ps,
+		lite:           noncer,
+		liteActorCache: must.One(lru.New[nonceCacheKey, result.Result[*types.Actor]](nonceCacheSize)),
+	}
 }
 
 func (mpp *mpoolProvider) IsLite() bool {
 	return mpp.lite != nil
 }
 
-func (mpp *mpoolProvider) getActorLite(addr address.Address, ts *types.TipSet) (*types.Actor, error) {
+func (mpp *mpoolProvider) getActorLite(addr address.Address, ts *types.TipSet) (act *types.Actor, err error) {
```

Comment:
you're not using `act` or `err` below your `defer`, they're always going to be unset as far as I can see; what am I missing?

Reply:
Nope, those values are set on return and accessible in defer. See https://go.dev/play/p/4qc2WmSDnZw

Reply:
(iiuc this behavior is actually the reason why named returns were conceived of - so that you can access return vals in defer)

Reply:
that's wild, didn't know that! thanks